src/BSON/ArrayList.stub.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * @generate-class-entries static
+ * @generate-function-entries static
+ */
+
+namespace MongoDB\BSON;
+
+final class ArrayList implements \IteratorAggregate
+{
+    private function __construct() {}
+
+    final static public function fromPHP(array $value): ArrayList {}
```

Comment:
If the class itself is final, is there any reason to make the methods final as well? Or is this just defensive in case the `final` class modifier is removed down the line?

Looking at other stubs (e.g. Binary), we don't use `final` for methods.

Reply:
Most (if not all) our classes use the `final` keyword for methods even when the class is already marked `final`. The old arginfo structures always used the `ZEND_ACC_FINAL` flag for methods despite the class having the flag as well. I'm not opposed to leaving them for convenience, even though they might be re-added in the (unlikely) event that we make a class non-final. If you want to drop them, I'd do so in a separate PR for all final classes.

Reply:
Oh, I was mistaken. The declarations in Binary were for consts and the methods therein do use `final`. Please disregard my original comment.